axelor-stock/src/main/resources/views/StockMove.xml

```diff
@@ -253,13 +253,14 @@
                 </field>
             </panel>
             <field name="filterOnAvailableProducts" widget="boolean-switch" onChange="action-stock-move-method-update-stock-move-line"/>
-			<panel name="shipmentInformationsPanel" title="Shipment Informations" showIf="typeSelect == 3">
+			<panel name="shipmentInformationsPanel" title="Shipment Informations" showIf="typeSelect == 3 &amp;&amp; !isReversion">
 				<panel name="SupplierShipmentDatePanel" itemSpan="12">
-					<field name="supplierShipmentDate" requiredIf="stockIsRequiredShipmentSupplierDetails &amp;&amp; statusSelect == 3"/>
+					<field name="supplierShipmentDate" requiredIf="$isRequiredShipmentSupplierDetails &amp;&amp; statusSelect == 3"/>
 					<label css="label-danger" title="Required" showIf="!supplierShipmentDate &amp;&amp; statusSelect == 3"/>
+					<field name="$isRequiredShipmentSupplierDetails" hidden="true"/>
 				</panel>
 				<panel name="SupplierShipmentRefPanel" itemSpan="12">
-					<field name="supplierShipmentRef" requiredIf="stockIsRequiredShipmentSupplierDetails &amp;&amp; statusSelect == 3"/>
+					<field name="supplierShipmentRef" requiredIf="$isRequiredShipmentSupplierDetails &amp;&amp; statusSelect == 3"/>
```

Comment:
you have to call an action-attrs to set a required condition on onLoad instead of dummy field